src/main/groovy/io/seqera/wave/service/builder/DockerBuildStrategy.groovy

```diff
@@ -54,7 +55,17 @@ class DockerBuildStrategy extends BuildStrategy {
     @Override
     void build(String jobName, BuildRequest req) {
 
-        final Path configFile = req.configJson ? req.workDir.resolve('config.json') : null
+        // save docker config for creds
+        Path configFile = null
+        if( req.configJson ) {
+            if ( req.formatSingularity() ) {
+                configFile = req.workDir.resolve('config.json')
+                Files.write(configFile, JsonOutput.prettyPrint(req.configJson).bytes, CREATE, WRITE, TRUNCATE_EXISTING)
+            } else {
+                configFile = req.workDir.resolve('config.json')
+            }
```

Comment:
Unify for docker and consider using [saveDockerAuth](https://github.com/seqeralabs/wave/blob/7d997c3935ee31006c804e7bb08aedf560ec6ed6/src/main/groovy/io/seqera/wave/service/job/JobHelper.groovy#L36-L36)